docker/Dockerfile

```diff
@@ -214,7 +214,14 @@ RUN source /opt/ros/$ROS_DISTRO/setup.bash && \
 ARG RMW_IMPLEMENTATION="rmw_cyclonedds_cpp"
 ENV RMW_IMPLEMENTATION=${RMW_IMPLEMENTATION}
 RUN source /opt/ros/$ROS_DISTRO/setup.bash && \
-    if [[ "$ROS_VERSION" == "2" ]]; then \
+    if [[ "$RMW_IMPLEMENTATION" == "rmw_zenoh" ]]; then \
```

Comment:
Regarding your comment

> We could also, for example, use the script build.sh to explicitly exclude the combinations of rmw_zenoh and an unsupported distro. At the moment, this would probably result in an error during the installation of rmw_zenoh.

I would not add more functionality to `build.sh`. Instead, I'd suggest to extend this `if` to `if(zenoh && (iron || jazzy || rolling))`. That way, we would also automatically install zenoh from apt once available in the next distro.

Reply:
I have also thought about this, but this means that if you select zenoh and humble, for example, no error will occur. This may lead you to think that zenoh is installed in the container, but it is not.

Reply:
Yes, valid point, then just add another if with an error message for Humble? Clunky, but safe, and better than to start modifying the other script, in my opinion.

Reply:
Decision: we keep it like this, `rmw_zenoh` is released as an apt package anyway